bootstrap.sh

```diff
@@ -85,6 +85,41 @@ if [[ ${master_settings_missing} == "true" ]]; then
   done
 fi
 
+# Create Elasticsearch snapshot repository
+if ! curl localhost:9200/_snapshot?pretty | grep $BACKUP_REPOSITORY; then
+
+  curl -XPUT localhost:9200/_snapshot/$BACKUP_REPOSITORY?pretty -H 'Content-Type: application/json' --data-binary @- << EOF
```

Comment:
Looks like this heredoc thingie is getting thrown away in cloud-init interpretation of script (I have changed calling to `bash -x /tmp/bootstrap.sh`) : 

Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: /tmp/bootstrap.sh: line 59: kill: (1798) - No such process
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + [[ false == \t\r\u\e ]]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: ++ curl -f localhost:9200/_snapshot/terraform-test-hejda-backups/
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:                                  Dload  Upload   Total   Spent    Left  Speed
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: [149B blob data]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + [[ ! -n '' ]]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + curl -XPUT 'localhost:9200/_snapshot/terraform-test-hejda-backups?pretty' -H 'Content-Type: application/json' --data-binary @-
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:                                  Dload  Upload   Total   Spent    Left  Speed
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: [149B blob data]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: ++ curl -f localhost:9200/_slm/policy/nightly-snapshots/
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:                                  Dload  Upload   Total   Spent    Left  Speed
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: [149B blob data]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + [[ ! -n '' ]]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + curl -XPUT 'localhost:9200/_slm/policy/nightly-snapshots?pretty' -H 'Content-Type: application/json' --data-binary @-
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]:                                  Dload  Upload   Total   Spent    Left  Speed
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: [149B blob data]
Aug 06 16:36:19 elasticsearch-prod-0 cloud-init[1506]: + systemctl enable elasticsearch.service

Reply:
Manual run in instance works, same code in script does not

Reply:
This is quite complicated to test. You always must taint ES instances and data disk. There were some false positives of working code when I've trashed only instances.

Reply:
Proposed heredoc (I guess from https://stackoverflow.com/questions/34847981/curl-with-multiline-of-json) didn't work for me either. Check my solution. It's a compromise but still has a little number of escape slashes.

There was also an issue with service startup. Elasticsearch was not available during backup settings `curl` calls.

Reply:
Works good :)